packages/next/src/server/app-render/dynamic-rendering.ts

```diff
@@ -180,9 +180,11 @@ export function trackDynamicFetch(
   store: StaticGenerationStore,
   expression: string
 ) {
-  if (store.prerenderState) {
-    postponeWithTracking(store.prerenderState, expression, store.urlPathname)
-  }
+  // If we aren't in a prerender, or we're in an unstable cache callback, we
+  // don't need to postpone.
+  if (!store.prerenderState || store.isUnstableCacheCallback) return
```

Comment:
This is the core change, during partial prerendering, it no longer marks the route as dynamic if we're in a `unstable_cache` callback.